tests/capi/unittest_capi_service_extension.cc

```diff
@@ -0,0 +1,1180 @@
+
+  _extension_test_imgclf (handle);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Usage of ml-service extension API.
+ */
+TEST_REQUIRE_TFLITE (ml_service_extension, scenarioConfig2ImgClf)
+{
+  ml_service_h handle;
+  int status;
+
+  /* The configuration file includes model path only. */
+  g_autofree gchar *config = _get_config_path ("config_single_imgclf_file.conf");
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  _extension_test_imgclf (handle);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Usage of ml-service extension API.
+ */
+TEST_REQUIRE_TFLITE (ml_service_extension, scenarioConfig3ImgClf)
+{
+  ml_service_h handle;
+  int status;
+
+  /* The configuration file includes model path only. */
+  g_autofree gchar *config = _get_config_path ("config_pipeline_imgclf.conf");
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  _extension_test_imgclf (handle);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (ml_service_extension, createConfigInvalidParam01_n)
+{
+  ml_service_h handle;
+  int status;
+
+  status = ml_service_new (NULL, &handle);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (ml_service_extension, createConfigInvalidParam02_n)
+{
+  ml_service_h handle;
+  int status;
+
+  status = ml_service_new ("", &handle);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST_REQUIRE_TFLITE (ml_service_extension, createConfigInvalidParam03_n)
+{
+  int status;
+
+  g_autofree gchar *config = _get_config_path ("config_single_add.conf");
+
+  status = ml_service_new (config, NULL);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (ml_service_extension, createConfigInvalidParam04_n)
+{
+  ml_service_h handle;
+  int status;
+
+  /* The configuration file does not exist. */
+  g_autofree gchar *config = _get_config_path ("invalid_path.conf");
+
+  status = ml_service_new (config, &handle);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (ml_service_extension, createConfigInvalidParam05_n)
+{
+  ml_service_h handle;
+  int status;
+
+  /* The configuration file has invalid tensor information. */
+  g_autofree gchar *config = _get_config_path ("config_single_imgclf_invalid_info.conf");
+
+  status = ml_service_new (config, &handle);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (ml_service_extension, createConfigInvalidParam06_n)
+{
+  ml_service_h handle;
+  int status;
+
+  /* The configuration file has invalid type. */
+  g_autofree gchar *config = _get_config_path ("config_single_unknown_type.conf");
```

Comment:
It is not the newly added file name. Is it "config_unknown_type.conf"?

Reply:
Thanks, updated.